core/opendaq/opendaq/include/opendaq/instance_builder.h

```diff
@@ -0,0 +1,249 @@
+
+    // [returnSelf]
+    /*!
+     * @brief Sets the amount of worker threads in the scheduler of Instance. Ignored if a scheduler has already been set
+     * @param numWorkers The amount of worker threads in the scheduler of Instance. If @c is 0, then the maximum number of concurrent threads supported by the implementation is used.
+     */
+    virtual ErrCode INTERFACE_FUNC setSchedulerWorkerNum(SizeT numWorkers) = 0;
+
+    /*!
+     * @brief Gets the amount of worker threads in the scheduler of Instance.
+     * @param[out] numWorkers The amount of worker threads in the scheduler of Instance.
+     */
+    virtual ErrCode INTERFACE_FUNC getSchedulerWorkerNum(SizeT* numWorkers) = 0;
+
+    // [returnSelf]
+    /*!
+     * @brief Sets the scheduler of Instance
+     * @param moduleManager The scheduler of Instance
+     */
+    virtual ErrCode INTERFACE_FUNC setScheduler(IScheduler* scheduler) = 0;
+
+    /*!
+     * @brief Gets the scheduler of Instance
+     * @param[out] moduleManager The scheduler of Instance
+     */
+    virtual ErrCode INTERFACE_FUNC getScheduler(IScheduler** scheduler) = 0;
+
+    // [returnSelf]
+    /*!
+     * @brief Sets the virtual Client as the default root device with a set name. If RootDevice has not been set, the Instance root device will be set as a virtual client.
+     * @param rootDevice The default root device name
+     */
+    virtual ErrCode INTERFACE_FUNC setDefaultRootDeviceLocalId(IString* localId) = 0;
+
+    /*!
+     * @brief Gets the default root device name
+     * @param[out] rootDevice The default root device name
+     */
+    virtual ErrCode INTERFACE_FUNC getDefaultRootDeviceLocalId(IString** localId) = 0;
+
+    // [returnSelf]
+    /*!
+     * @brief Sets the connection string for the root device of Instance.
+     * @param connectionString The connection string for the default root device of Instance
+     */
+    virtual ErrCode INTERFACE_FUNC setRootDevice(IString* connectionString) = 0;
+
+    /*!
+     * @brief Gets the connection string for the default root device of Instance.
+     * @param[out] rootDevice The connection string for the default root device of Instance
+     */
+    virtual ErrCode INTERFACE_FUNC getRootDevice(IString** connectionString) = 0;
+
+    // [returnSelf]
+    /*!
+     * @brief Sets the default device info of Instance. If device info has been set, method getInfo of Instance will return set device info if Root Device has not been set
+     * @param deviceInfo The device info of the default device of Instance
+     */
+    virtual ErrCode INTERFACE_FUNC setDefaultRootDeviceInfo(IDeviceInfo* deviceInfo) = 0;
+
+    /*!
+     * @brief Gets the default device info of Instance
+     * @param defaultDevice The default device info of Instance
+     */
+    virtual ErrCode INTERFACE_FUNC getDefaultRootDeviceInfo(IDeviceInfo** deviceInfo) = 0;
+};
+/*!@}*/
+
+/*!
+ * @brief Creates a InstanceBuilder with no parameters configured.
+ */
```

Comment:
This comment should be in a Doxygen group

Reply:
there is a comment in core\opendaq\opendaq\include\opendaq\instance_factory.h